firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLitePersistence.java

```diff
@@ -455,6 +459,122 @@ private Cursor startQuery() {
     }
   }
 
+  /**
+   * Encapsulates a query whose parameter list is so long that it might exceed SQLite limit.
+   *
+   * <p>SQLite limits maximum number of host parameters to 999 (see
+   * https://www.sqlite.org/limits.html). This class wraps most of the messy details of splitting a
+   * large query into several smaller ones.
+   */
+  static class LongQuery {
+    private final SQLitePersistence db;
+    private final String head;
+    private final String tail;
+    private final List<Object> argsHead;
+    private final List<Object> allArgs;
+
+    private int queriesPerformed = 0;
+    private Iterator<Object> argsIter;
+
+    private static final int LIMIT = 900;
+
+    /**
+     * Creates a new {@code LongQuery} with parameters that describe a template for creating each
+     * subquery.
+     *
+     * <p>Each subquery will have the following form:
```

Comment:
All of this explanatory text is great and I wonder if it'd make sense to move some of it to the class instead of on the constructor, since I think it applies to both constructors (and it would have made it easier for me to understand the class members above).